File: web-service/api-project/src/main/java/io/ehdev/conrad/service/api/service/model/permissions/PermissionGrant.java
package io.ehdev.conrad.service.api.service.model.permissions;

import com.fasterxml.jackson.annotation.JsonProperty;

public class PermissionGrant {

    @JsonProperty("username")
    String username;

    @JsonProperty("permission")
    String permission;

    public PermissionGrant(String username, String permission) {
        this.username = username;
        this.permission = permission;
    }

    public PermissionGrant() {
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPermission() {
        return permission;
    }

    public void setPermission(String permission) {
        this.permission = permission;
    }
}

File: api/rest-model/src/main/kotlin/io/ehdev/conrad/model/permission/PermissionGrant.kt
package io.ehdev.conrad.model.permission

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyDescription

data class PermissionGrant(
    @JsonPropertyDescription("Name of the user to grant the permission to.")
    @JsonProperty("username")
    val username: String,

    @JsonPropertyDescription("Type of permission to grant access to")
    @JsonProperty("permission")
    val permission: PermissionDefinition) {

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    enum class PermissionDefinition {
        NONE,
        READ,
        WRITE,
        ADMIN
    }

}
